cmd/kubeapps-apis/server/server.go

```diff
@@ -88,3 +109,88 @@ type gwHandlerArgs struct {
 	addr        string
 	dialOptions []grpc.DialOption
 }
+
+// registerPlugins opens each plugin, looks up the register function and calls it with the registrar.
+func registerPlugins(pluginPaths []string, grpcReg grpc.ServiceRegistrar, gwArgs gwHandlerArgs) error {
+	for _, pluginPath := range pluginPaths {
+		p, err := plugin.Open(pluginPath)
+		if err != nil {
+			return fmt.Errorf("unable to open plugin %q: %w", pluginPath, err)
+		}
+
+		if err = registerGRPC(p, pluginPath, grpcReg); err != nil {
+			return err
+		}
+
+		if err = registerHTTP(p, pluginPath, gwArgs); err != nil {
+			return err
+		}
+
+		log.Infof("Successfully registered plugin %q", pluginPath)
+	}
+	return nil
+}
+
+// registerGRPC finds and calls the required function for registering the plugin for the GRPC server.
+func registerGRPC(p *plugin.Plugin, pluginPath string, registrar grpc.ServiceRegistrar) error {
+	grpcRegFn, err := p.Lookup(grpcRegisterFunction)
```

Comment:
Interesting usage of reflection here to be able to register the plugging in both grpc and HTTP server.

Reply:
This is how the `plugin` package recommends calling the functions. The only non-standard bit is the `dummyFn` to try to get a useful error message.